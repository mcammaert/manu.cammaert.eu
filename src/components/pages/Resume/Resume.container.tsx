import React from 'react';

import {
  getAllJobs,
  getPersonal,
  getAllSkills,
  getAllTechnical,
  getAllSoftware,
  getAllEducation,
  getAllTraining,
  getAllLanguages,
  getAllHobbies,
} from 'api/resume';
import Resume from './Resume';

import { ResumeContainerProps } from './Resume.types';

const ResumeContainer: React.FC<ResumeContainerProps> = () => {
  const jobs = getAllJobs();
  const personal = getPersonal();
  const skills = getAllSkills();
  const technical = getAllTechnical();
  const software = getAllSoftware();
  const education = getAllEducation();
  const training = getAllTraining();
  const languages = getAllLanguages();
  const hobbies = getAllHobbies();

  return (
    <Resume
      jobs={jobs || []}
      personal={personal}
      skills={skills}
      technical={technical}
      software={software}
      education={education}
      training={training}
      languages={languages}
      hobbies={hobbies}
    />
  );
};

export default ResumeContainer;
