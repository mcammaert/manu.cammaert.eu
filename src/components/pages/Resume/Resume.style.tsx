import styled from 'styled-components';
import { modularScale } from 'polished';

export const Resume = styled.div`
  position: absolute;
`;

export const LabelList = styled.ul`
  padding: 0;
`;

export const LabelListItem = styled.li`
  display: inline-block;
  margin-bottom: ${modularScale(-3)};
  line-height: ${modularScale(2)};
  margin-right: ${modularScale(-3)};
`;

export const Jobs = styled.div`
  margin-top: ${modularScale(1)};
`;

export default { Resume, LabelList, LabelListItem, Jobs };
