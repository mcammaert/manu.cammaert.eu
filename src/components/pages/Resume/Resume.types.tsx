import { Jobs } from 'api/resume';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface ResumeContainerProps {}

export interface ResumeProps {
  jobs: Jobs;
}
