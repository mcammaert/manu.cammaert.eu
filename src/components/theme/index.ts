import { createGlobalStyle } from 'styled-components';
import 'normalize.css';
import { colors } from './colors';

export interface GlobalStyleProps {
  readonly light: boolean;
}

// const backgroundColor = (location: string) => (location === '/' ? colors.oxfordBlue : colors.fuzzyWuzzy);

export const GlobalStyle = createGlobalStyle<GlobalStyleProps>`
  body {
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    font-family: 'Nunito Sans', sans-serif;
    width: 100vw;
    background: ${({ light }) => (light ? colors.white : colors.black)};
    background-attachment: fixed;    
    color: ${({ light }) => (light ? colors.black : colors.snow)};
    transition: all 800ms;
    a {
      color: ${({ theme }) => theme.colors.secondary};
    }
  }
`;

export const lightTheme = {
  margin: 15,
  colors: {
    primary: colors.charcoal,
    primaryInvert: colors.snow,
    secondary: colors.primary60,
    secondaryInvert: colors.white,
    tertiary: colors.secondary80,
    tertiaryInvert: colors.white,
    copy: colors.charcoal,
    copyInvert: colors.white,
  },
};

export const darkTheme = {
  margin: 15,
  colors: {
    primary: colors.snow,
    primaryInvert: colors.black,
    secondary: colors.primary30,
    secondaryInvert: colors.primary90,
    tertiary: colors.secondary30,
    tertiaryInvert: colors.secondary80,
    copy: colors.white,
    copyInvert: colors.charcoal,
  },
};
