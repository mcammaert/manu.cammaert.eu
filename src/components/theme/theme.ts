import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  body {
    background-color: #ccc;
  }
`;

const theme = {
  main: 'mediumseagreen',
};

export default theme;
