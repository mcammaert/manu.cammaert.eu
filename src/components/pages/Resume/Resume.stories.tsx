import React from 'react';

import { storiesOf } from '@storybook/react';

import Resume from './Resume';

storiesOf('Resume', module).add('Resume', () => {
  return (
    <Resume
      jobs={[]}
      personal={{
        name: 'name',
        nationality: 'nationality',
        birthDate: '2000-01-01',
        address: 'address',
        email: 'email',
        phone: '+32 485 00 00 00',
      }}
      skills={[]}
      technical={[]}
      software={[]}
      education={[]}
      training={[]}
      languages={[]}
      hobbies={[]}
    />
  );
});
