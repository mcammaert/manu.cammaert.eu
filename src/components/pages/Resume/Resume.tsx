import React from 'react';

import { Default } from 'components/templates/Default';

import { ResumeProps } from './Resume.types';

import S from './Resume.style';

const Resume: React.FC<ResumeProps> = () => (
  <S.Resume>
    <Default>
      <p>Curriculum Vitae</p>
    </Default>
  </S.Resume>
);

export default Resume;
