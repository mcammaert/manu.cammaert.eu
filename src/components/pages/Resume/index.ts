import * as S from './Resume.style';

export { default as Resume } from './Resume';
export * from './Resume.types';
export { S };
