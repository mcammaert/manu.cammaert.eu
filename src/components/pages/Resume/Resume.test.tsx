import React from 'react';

import { cleanup } from '@testing-library/react';
import 'jest-styled-components';

import { renderComponent } from 'testing/react-testing-library';
import { lightTheme } from 'components/theme';

import Resume from './Resume';

afterEach(cleanup);

describe('Resume', () => {
  describe('default props', () => {
    it('should match the snapshot', () => {
      const component = (
        <Resume
          jobs={[]}
          personal={{
            name: 'name',
            nationality: 'nationality',
            birthDate: '2000-01-01',
            address: 'address',
            email: 'email',
            phone: '+32 485 00 00 00',
          }}
          skills={[]}
          technical={[]}
          software={[]}
          education={[]}
          training={[]}
          languages={[]}
          hobbies={[]}
        />
      );
      const { container } = renderComponent({ theme: lightTheme, component });
      expect(container).toMatchSnapshot();
    });
  });
});
