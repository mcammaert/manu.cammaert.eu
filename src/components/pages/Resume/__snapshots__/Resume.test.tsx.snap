// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resume default props should match the snapshot 1`] = `
.c1 {
  width: 100vw;
}

.c9 {
  position: absolute;
  font-size: 0.7501875468867217em;
  padding-top: 5.610232974924882em;
  padding-bottom: 5.610232974924882em;
  width: 100vw;
}

.c11 {
  color: rgba(38,40,48,0.45);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 1.333em;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 0;
}

.c3 {
  width: 100vw;
  max-width: 800px;
  padding-left: 0.9375rem;
  padding-right: 0.9375rem;
  margin-top: 0;
  margin-bottom: 0;
}

.c4 {
  font-size: 1.776889em;
  line-height: 1.333em;
  font-weight: 200;
  color: #262830;
  margin: 0;
  margin-bottom: 0.4221915645190677em;
}

.c6 {
  font-size: 1em;
  line-height: 1.333em;
  font-weight: 200;
  color: #262830;
  margin: 0;
  margin-bottom: 0.4221915645190677em;
}

.c7 {
  width: 100%;
  margin-bottom: 1em;
}

.c7 th {
  font-size: 0.7501875468867217em;
  text-align: left;
  padding: 0.5627813555039173em;
  font-weight: 400;
  text-transform: uppercase;
}

.c7 td {
  font-size: 1em;
  border-top: 1px solid rgba(38,40,48,0.08);
  padding: 0.5627813555039173em;
  font-weight: 400;
}

.c0 {
  position: absolute;
}

.c5 {
  padding: 0;
}

.c8 {
  margin-top: 1.333em;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <main>
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <h2
              class="c4"
              size="2"
            >
              Persoonlijke gegevens
            </h2>
          </div>
        </div>
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <h2
              class="c4"
              size="2"
            >
              Vaardigheden
            </h2>
            <ul
              class="c5"
            />
          </div>
        </div>
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <h2
              class="c4"
              size="2"
            >
              Technologiën
            </h2>
            <ul
              class="c5"
            />
          </div>
        </div>
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <h2
              class="c4"
              size="2"
            >
              Software
            </h2>
            <ul
              class="c5"
            />
          </div>
        </div>
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <h2
              class="c4"
              size="2"
            >
              Opleidingen en training
            </h2>
            <h3
              class="c6"
            >
              Opleidingen
            </h3>
            <ul
              class="c5"
            />
            <h3
              class="c6"
            >
              Training
            </h3>
            <ul
              class="c5"
            />
          </div>
        </div>
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <h2
              class="c4"
              size="2"
            >
              Talen
            </h2>
            <table
              class="c7"
            >
              <thead>
                <tr>
                  <th
                    style="width: 20%;"
                  >
                    Taal
                  </th>
                  <th>
                    Gesproken
                  </th>
                  <th>
                    Geschreven
                  </th>
                </tr>
              </thead>
              <tbody />
            </table>
          </div>
        </div>
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <h2
              class="c4"
              size="2"
            >
              Hobbies
            </h2>
            <ul
              class="c5"
            />
          </div>
        </div>
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <h2
              class="c4"
              size="2"
            >
              Loopbaan
            </h2>
          </div>
        </div>
        <div
          class="c8"
        />
      </main>
      <footer
        class="c9"
      >
        <div
          class="c10"
        >
          <div
            class="c3"
          >
            <div
              class="c11"
            >
              Copyright © 
              2019
               Manu Cammaert, alle rechten voorbehouden.
            </div>
          </div>
        </div>
      </footer>
    </div>
  </div>
</div>
`;
